src/header.tsx

```diff
@@ -38,7 +38,11 @@ import { GripVerticalIcon, ListIcon } from "@patternfly/react-icons";
 
 const _ = cockpit.gettext;
 
-export const FilesCardHeader = ({ currentFilter, onFilterChange, isGrid, setIsGrid, sortBy, setSortBy }) => {
+export const FilesCardHeader = ({ currentFilter, onFilterChange, isGrid, setIsGrid, sortBy, setSortBy }:
+    { currentFilter: string, onFilterChange: (_event: React.FormEvent<HTMLInputElement>, value: string) => void,
+      isGrid: boolean, setIsGrid: React.Dispatch<React.SetStateAction<boolean>>,
+      sortBy: string, setSortBy: React.Dispatch<React.SetStateAction<string>>
```

Comment:
I guess we all agree that this (1) looks sad, and (2) is currently unavoidable?

I have a slight preference to putting each attribute in its own line, makes it easier to read. But no PR blocker.

Is it possible somehow to add a shortcut for `React.Dispatch<React.SetStateAction<T>>` ? We'll need that *a lot*, so some `setState<T>` alias would be nice.

Reply:
Then we would need to import that in almost all files, we can but it feels a bit off.